src/main/java/duke/command/HelpCommand.java

```diff
@@ -0,0 +1,75 @@
+package duke.command;
+
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+public class HelpCommand extends Command {
+	private boolean isCommandSpecific;
+	private String inputCommand;
+	
+	public HelpCommand(boolean isCommandSpecific, String inputCommand) {
+		super("Help", new Task());
+		this.isCommandSpecific = isCommandSpecific;
+		this.inputCommand = inputCommand;
+	}
+	
+	
+	@Override
+	public String execute(TaskList list, Ui ui, Storage storage) {
+		String helpOutput = null;
+		if (isCommandSpecific) {
+			helpOutput = executeSpecificCommand(inputCommand);
+		} else {
+			helpOutput = executeGeneral();
+		}
+		return helpOutput;
+	}
+	
+	private String executeSpecificCommand(String commandInterested) {
```

Comment:
what is the difference between the execute() and executeSpecificCommand() methods? is it possible to eliminate one of those, or improve the abstraction of helper methods for execute()?